firmware/controllers/math/pid.cpp

```diff
@@ -151,3 +145,61 @@ void Pid::showPidStatus(Logging *logging, const char*msg) {
 			iTerm, dTerm);
 
 }
+
+void Pid::updateITerm(float value) {
+	iTerm += value;
+	/**
+	 * If we have exceeded the ability of the controlled device to hit target, the I factor will keep accumulating and approach infinity.
+	 * Here we limit the I-term #353
+	 */
+	if (iTerm > pid->maxValue * 100)
+		iTerm = pid->maxValue * 100;
+
+	// this is kind of a hack. a proper fix would be having separate additional settings 'maxIValue' and 'minIValye'
+	if (iTerm < -pid->maxValue * 100)
+		iTerm = -pid->maxValue * 100;
+}
+
+
+PidCic::PidCic() {
+	// call our derived reset()
```

Comment:
would it be nicer to invoke parent constructor?

Reply:
Unfortunately, in C++ the  polymorphic inheritence is limited, unlike Java/C#. 
The parent constructor doesn't know how to call the virtual reset() method, because vtbl is not created jet...
https://isocpp.org/wiki/faq/strange-inheritance#calling-virtuals-from-ctors